### DISCUSSION TITLE
mp3 is saved as mpeg

### DISCUSSION THREAD

**Discussion by intermezzzo** [aclap-dev/video-downloadhelper#76] [Help and Q&A]
My to download file is an MP3 file. Now (in Edge 8.2.0.20) it is suddenly saved without any intervention to mpeg. 
I want my mp3 as before. 
How to get this back?

**Comment by GrampaWildWilly** [comment]
I have noticed this in Firefox as well.  I just use the VDH Download function instead of Quick Download.  That lets me change the file name before letting VDH start its download.  I just change .mpeg to .mp3 & it works fine.  But I agree that it should start by naming the default file mp3 & not mpeg.  On the other hand, VLC plays the mpeg file just fine as audio only, so this is more of a cosmetic fix than anything else.  And in case you didn't figure it out from the way I'm saying this, I don't work for VDH.  I'm just another user.

**Reply by intermezzzo** [reply]
The mp3 file must be played by good old winamp, that did not tke the mpeg file.!

**Reply by GrampaWildWilly** [reply]
Hmmm........  Interesting.  Maybe that's yet another reason to use VLC instead.

**Reply by GrampaWildWilly** [reply]
In my experience, VLC doesn't pay any attention at all to the file name, in particular, the file extension.  It looks inside the file & figures out what it really is, then plays it accordingly, assuming it's a proper audio/visual file.

**Reply by intermezzzo** [reply]
I my case this it is not possible to use VLC for the application. So I hope that I can get the not conveted file gain back as before with VDH! The FireFox trick does not work in Edge. I cannot get the save as dialog anymore.

**Reply by GrampaWildWilly** [reply]
On 2 separate paths . . .

After VDH downloads it as .mpeg, I have found that I can just rename the file to .mp3.  Does that not satisfy winamp?

Unable to get the non-Quick Download dialog?  That sounds like it should be a separate discussion.

**Reply by intermezzzo** [reply]
No, the file format is wrong. It hicks in WinAmp. It results in let say, remarkble playing.
Indeed, save as dialog does not appear in any setting.

**Comment by paulrouget** [comment]
I need a link to a page with a MP3 to test.

**Reply by intermezzzo** [reply]
I wrote a 'manual' here: https://cardbox3.stackstorage.com/s/uooORKP3ZS7n2pgA

**Reply by paulrouget** [reply]
For reference, url is https://www.nporadio1.nl/

**Reply by GrampaWildWilly** [reply]
I went there & the VDH menu was empty.  So I clicked on LAATSTE JOURNAAL as you directed & got this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6ba3cc78-ee27-4b5a-bcc0-68cb085807d3)

So I clicked on that one choice offered in the VDH menu.  I got this:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/939fc539-5cee-472e-82b5-565c80c1889f)

I got the file selection dialog.  But then, I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0.1 64-bit, licensed VDH 8.2.0.20a1 beta, CoApp 2.0.17.  Different browser, maybe different operating system.  You didn't say what you're running.  I thought the m2ts file extension was not right.  So I changed it:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/26319540-786b-4307-a627-16ccd633fab9)

Pretty soon it told me this:

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/cbe68d96-9da2-42c8-b881-4c74a783153c)

Not good.  I clicked the red dot:

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/412926d5-3eaa-42de-bd96-a2fae49b0593)

Oh really.  Clicking those Details got me this:

```
Exit code: 1

moz-extension://9d469d39-7727-4f85-8beb-0761511221b9/background/main.js:1
receive@moz-extension://9d469d39-7727-4f85-8beb-0761511221b9/background/main.js:1:13157
Yp</callCatchAppNotFound/</<@moz-extension://9d469d39-7727-4f85-8beb-0761511221b9/background/main.js:1:42562
```

I hope that means something to Paul because it means squat to me.

Then I clicked the PLAY button & I got this:

![#06](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/01939c40-1c44-4ac4-baa8-b298cd2792a2)

That list of variants kept growing as long as I kept the thing playing, which wasn't long since I don't speak Dutch.

Something is rotten in the state of Denmark.  Er . . . I mean, the Netherlands.

**Reply by GrampaWildWilly** [reply]
On a hunch, I went back & tried the download again, this time leaving the file extension as m2ts.  Same result.  Same error code 1, same Details.

**Reply by intermezzzo** [reply]
After pushed the 'button' and while the news is playing, I get this which is ok. 
![image](https://github.com/aclap-dev/video-downloadhelper/assets/159853950/f2886222-0078-4cd7-8aa4-b5553614d272)
The mp3 = about 2 minutes. Click on it results without intercenption a mpeg file in the VHD download folder. If I setup VHD in downloadprocessor in "ask" nothing is asked. Win 10, pro, Edge as wel as Chrome, all updated.  Prior versions of VHD were ok. Problem is new in this update

**Reply by GrampaWildWilly** [reply]
Yeah, now I'm getting that as well.  It wasn't doing that earlier.  I'm not getting the ever expanding list of small mp4s now.  It's still downloading as mpeg instead of mp3, though.  But Paul says below he's working on that.

**Comment by paulrouget** [comment]
I'm working on that. Got a fix. Need proper testing then I'll publish a new version.

**Reply by intermezzzo** [reply]
Many thanks!

**Comment by paulrouget** [comment]
See https://github.com/aclap-dev/video-downloadhelper/discussions/82

**Reply by intermezzzo** [reply]
Wow!!!!
